### PR TITLE
Fix export job dataset

### PR DESCRIPTION
**Pull request by Marishka17**
### Motivation and context

In the previous implementation, the annotation file for the job contained information related to the task. Zip archive with the dataset contained all images of the task. This PR fixed this issue. 
### How has this been tested?

Manually
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly~~
- [ ] I have added tests to cover my changes
~~- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
~~- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by zhiltsov-max** [comment]
When I tried to export the 2nd job in a task with 6 images and 3 jobs (2 img each) as `Segmentation mask 1.1`, I've got the following error:
```
Traceback (most recent call last):
  File "/home/max/projects/cvat/venv/lib/python3.9/site-packages/rq/worker.py", line 936, in perform_job
    rv = job.perform()
  File "/home/max/projects/cvat/venv/lib/python3.9/site-packages/rq/job.py", line 684, in perform
    self._result = self._execute()
  File "/home/max/projects/cvat/venv/lib/python3.9/site-packages/rq/job.py", line 690, in _execute
    return self.func(*self.args, **self.kwargs)
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/views.py", line 108, in export_job_annotations
    return export(dst_format,job_id=job_id, server_url=server_url, save_images=False)
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/views.py", line 81, in export
    export_fn(db_instance.id, temp_file, dst_format,
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/task.py", line 716, in export_job
    job.export(f, exporter, host=server_url, save_images=save_images)
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/task.py", line 561, in export
    exporter(dst_file, job_data, **options)
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/formats/registry.py", line 37, in __call__
    f_or_cls(*args, **kwargs)
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/formats/mask.py", line 28, in _export
    apply_colormap=True, label_map=make_colormap(instance_data))
  File "/home/max/projects/cvat/cvat/apps/dataset_manager/formats/utils.py", line 53, in make_colormap
    labels = [label for _, label in instance_data.meta[instance_name]['labels']]
KeyError: 'task'
```

This also applies to other segmentation formats (`CamVid`, `Cityscapes`). `CVAT for images` and `COCO` seem to work.

**Comment by Marishka17** [comment]
> When I tried to export the 2nd job in a task with 6 images and 3 jobs (2 img each) as `Segmentation mask 1.1`, I've got the following error:
> 
> ```
> Traceback (most recent call last):
>   File "/venv/lib/python3.9/site-packages/rq/worker.py", line 936, in perform_job
>     rv = job.perform()
>   File "/venv/lib/python3.9/site-packages/rq/job.py", line 684, in perform
>     self._result = self._execute()
>   File "/venv/lib/python3.9/site-packages/rq/job.py", line 690, in _execute
>     return self.func(*self.args, **self.kwargs)
>   File "/cvat/apps/dataset_manager/views.py", line 108, in export_job_annotations
>     return export(dst_format,job_id=job_id, server_url=server_url, save_images=False)
>   File "/cvat/apps/dataset_manager/views.py", line 81, in export
>     export_fn(db_instance.id, temp_file, dst_format,
>   File "/cvat/apps/dataset_manager/task.py", line 716, in export_job
>     job.export(f, exporter, host=server_url, save_images=save_images)
>   File "/cvat/apps/dataset_manager/task.py", line 561, in export
>     exporter(dst_file, job_data, **options)
>   File "/cvat/apps/dataset_manager/formats/registry.py", line 37, in __call__
>     f_or_cls(*args, **kwargs)
>   File "/cvat/apps/dataset_manager/formats/mask.py", line 28, in _export
>     apply_colormap=True, label_map=make_colormap(instance_data))
>   File "/cvat/apps/dataset_manager/formats/utils.py", line 53, in make_colormap
>     labels = [label for _, label in instance_data.meta[instance_name]['labels']]
> KeyError: 'task'
> ```

Thanks, this issue has been fixed

> This also applies to other segmentation formats (`CamVid`, `Cityscapes`). `CVAT for images` and `COCO` seem to work.
> 
> Also, don't you think the old button for task export in the menu of the annotation page should be restored? Users may expect it to be there, and it was quite useful there. image

I guess that button for to export task should be on the task detail page or on the tasks page. Let's wait for customer requests.

> Also, consider adding a job menu button here: image

We have a button to export jobs in the annotation view and on jobs page. Let's wait for customer requests.

File: cvat/apps/dataset_manager/bindings.py [hunk 1]
```diff
@@ -1108,24 +1252,25 @@ def map_label(name, parent=''): return label_cat.find(name, parent)[0]
         return convert_cvat_anno_to_dm(cvat_frame_anno, label_attrs, map_label)
 
 
-class CvatTaskDataExtractor(dm.SourceExtractor, CVATDataExtractorMixin):
-    def __init__(self, task_data, include_images=False, format_type=None, dimension=DimensionType.DIM_2D):
+class CvatTaskOrJobDataExtractor(dm.SourceExtractor, CVATDataExtractorMixin):
+    def __init__(self, instance_data: Union[TaskData, JobData], include_images=False, format_type=None, dimension=DimensionType.DIM_2D):
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/bindings.py:1256]: Wouldn't it make sense to use `CommonData` as an annotation here? I assume that was the whole purpose of the class, to serve as a common interface.

Same in many other cases below where `TaskData, JobData` or `JobData, TaskData` is used.

File: cvat/apps/dataset_manager/bindings.py [hunk 2]
```diff
@@ -605,7 +576,180 @@ def match_frame_fuzzy(self, path):
...
+        return range(segment.start_frame, segment.stop_frame + 1)
+
+    @property
+    def start(self):
+        segment = self._db_job.segment
+        return segment.start_frame
+
+    @property
+    def offset(self):
+        return self.start
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/bindings.py:659]: Why do both `offset` and `start` need to exist? Seems like they are the same thing.

**Review by SpecLad — APPROVED** [review]

**Pull request merged**